vmdb/app/models/mixins/authentication_mixin.rb

```diff
@@ -74,8 +91,19 @@ def update_authentication(data, options = {})
     data.each_pair do |type, value|
       cred = self.authentication_type(type)
       current = {:new => nil, :old => nil}
-      current[:new] = {:user => value[:userid], :password => value[:password]} unless value[:userid].blank?
-      current[:old] = {:user => cred.userid, :password => cred.password} if cred
+
+      if value[:auth_key]
+        cred = self.auth_user_keypair(type)
+        # TODO(lsmola) figure out if there is a better way. Password field is replacing \n with \s, I need to replace
+        # them back
+        fixed_auth_key = value[:auth_key].gsub(/-----BEGIN\sRSA\sPRIVATE\sKEY-----/, '')
+        fixed_auth_key = fixed_auth_key.gsub(/-----END\sRSA\sPRIVATE\sKEY-----/, '')
+        fixed_auth_key = fixed_auth_key.gsub(/\s/, "\n")
+        value[:auth_key] = '-----BEGIN RSA PRIVATE KEY-----' + fixed_auth_key + '-----END RSA PRIVATE KEY-----'
```

Comment:
@jrafanie I understand. But the whole code L91-L99 will be deleted, when PR for file upload for private key will be implemented. So there should be no need to refactor this now (by every refactoring I need to rebase like 4 PRs dependent on this :-) )

This also should not affect anything else, since auth_key wasn't passed here before.